src/PostHog/ErrorTracking/ExceptionPropertiesBuilder.cs

```diff
@@ -0,0 +1,145 @@
+                ["value"] = ex.Message,
+                ["mechanism"] = new Dictionary<string, object>
+                {
+                    ["type"] = "generic",
+                    ["handled"] = true,
+                    ["source"] = "",
+                    ["synthetic"] = false
+                },
+                ["stacktrace"] = new Dictionary<string, object>
+                {
+                    ["frames"] = BuildStackFrameList(ex),
+                    ["type"] = "raw"
+                }
+            });
+
+            if (ex is AggregateException aex)
+            {
+                foreach (var inner in aex.Flatten().InnerExceptions.Reverse())
+                    if (inner is not null) stack.Push(inner);
+            }
+            else if (ex.InnerException is not null)
+            {
+                stack.Push(ex.InnerException);
+            }
+        }
+
+        return list;
+    }
+
+    private static List<Dictionary<string, object>> BuildStackFrameList(Exception exception)
+    {
+        var stackFrames = new List<Dictionary<string, object>>();
+        var stackTrace = new System.Diagnostics.StackTrace(exception, true);
+
+        foreach (var frame in stackTrace.GetFrames() ?? [])
+        {
+            var method = frame.GetMethod();
+            var fileName = frame.GetFileName();
+            var lineNumber = frame.GetFileLineNumber();
+            var columnNumber = frame.GetFileColumnNumber();
+
+            var frameDetails = new Dictionary<string, object>
+            {
+                ["platform"] = "custom",
+                ["lang"] = "dotnet",
+                ["filename"] = Path.GetFileName(fileName) ?? "",
+                ["abs_path"] = fileName ?? "",
+                ["function"] = method?.Name ?? "",
+                ["module"] = method?.DeclaringType?.FullName ?? "",
+                ["lineno"] = lineNumber,
+                ["colno"] = columnNumber
+            };
+
+            if (string.IsNullOrEmpty(fileName))
+            {
+                stackFrames.Add(frameDetails);
+                continue;
+            }
+
+            var sourceContext = BuildSourceCodeContext(fileName, lineNumber, columnNumber,
+                DEFAULT_MAX_LINES, DEFAULT_MAX_LENGTH);
+
+            if (!string.IsNullOrEmpty(sourceContext.ContextLine))
+            {
+                frameDetails["pre_context"] = sourceContext.PreContext ?? [];
+                frameDetails["context_line"] = sourceContext.ContextLine;
+                frameDetails["post_context"] = sourceContext.PostContext ?? [];
+            }
+
+            stackFrames.Add(frameDetails);
+        }
+
+        return stackFrames;
+    }
+
+    private static SourceCodeContext BuildSourceCodeContext(
+        string absolutePath,
+        int lineNumber,
+        int columnNumber,
+        int maxLines = 0,
+        int maxLength = 0)
+    {
+        var lines = File.ReadAllLines(absolutePath);
```

Comment:
This line could throw an exception for a variety of reasons. I recommend adding a method for reading the path safely and using that:

```csharp
private static string? ReadSourceCodeSafely(string absolutePath)
{
    if (string.IsNullOrEmpty(absolutePath) || !File.Exists(absolutePath))
        return null;

    try
    {
        return File.ReadAllLines(absolutePath);
    }
    catch (Exception ex) when (
        ex is IOException or UnauthorizedAccessException or NotSupportedException)
    {
        return null;
    }
}
```